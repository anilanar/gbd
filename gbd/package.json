{
  "name": "gbd",
  "version": "0.3.1-alpha.0",
  "description": "debug expressions with pretty printed output",
  "repository": "https://github.com/anilanar/gbd",
  "author": "Anil Anar <mail@anilanar.me>",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "MIT",
  "scripts": {
    "build": "tsc",
    "test": "exit 0",
    "lint": "exit 0"
  },
  "dependencies": {
    "pretty-format": "^25.3.0"
  }
}
