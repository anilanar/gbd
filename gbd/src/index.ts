import format from "pretty-format";

/**
 * Pretty prints a value and returns it unchanged.
 * @param val Value to debug.
 * @param expr Passed by the babel plugin.
 * @returns Value that was passed, unchanged.
 */
export function gbd<T>(val: T, expr?: string): T {
  if (expr === undefined) {
    console.error(
      "gdb :: No expression is found. Did you forget to use the babel plugin or the babel macro?"
    );
    return val;
  }
  console.debug(`> ${expr}\n= ${format(val)}`);
  return val;
}
