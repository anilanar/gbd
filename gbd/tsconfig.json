{
  "extends": "../tsconfig.shared.json",
  "compilerOptions": {
    "emitDeclarationOnly": false,
    "lib": ["ES5", "DOM"],
    "outDir": "dist",
    "rootDir": "src"
  },
  "files": ["src/index.ts"],
  "references": [{ "path": "../babel-transform-gbd" }]
}
