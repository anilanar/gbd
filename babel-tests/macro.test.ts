import pluginTester from "babel-plugin-tester";
import plugin from "babel-plugin-macros";

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: [
    `
      import { gbd } from 'gbd.macro'
      gbd(1 + 1);
    `,
    `
      import { gbd } from 'gbd.macro'
      gbd(new foo(1 + 1)[1].bar());
    `,
    `
      import { gbd } from 'gbd.macro'
      gbd({
        multi: 'foo',
        line: 'bar',
        obj: 'quu',
      });
    `,
    `
      import { gbd } from 'gbd.macro'
      function test() {
        gbd({
          multi: 'foo',
          line: 'bar',
          obj: 'quu',
        });
      }
    `,
  ],
});
