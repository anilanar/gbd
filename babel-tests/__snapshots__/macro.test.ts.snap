// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `

import { gbd } from 'gbd.macro'
gbd(1 + 1);

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";

_gbd(1 + 1, "1 + 1");


`;

exports[`macros 1. macros: 1. macros 2`] = `

import { gbd } from 'gbd.macro'
gbd(1 + 1);

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

(0, _gbd2.gbd)(1 + 1, "1 + 1");


`;

exports[`macros 2. macros: 2. macros 1`] = `

import { gbd } from 'gbd.macro'
gbd(new foo(1 + 1)[1].bar());

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";

_gbd(new foo(1 + 1)[1].bar(), "new foo(1 + 1)[1].bar()");


`;

exports[`macros 2. macros: 2. macros 2`] = `

import { gbd } from 'gbd.macro'
gbd(new foo(1 + 1)[1].bar());

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

(0, _gbd2.gbd)(new foo(1 + 1)[1].bar(), "new foo(1 + 1)[1].bar()");


`;

exports[`macros 3. macros: 3. macros 1`] = `

import { gbd } from 'gbd.macro'
gbd({
  multi: 'foo',
  line: 'bar',
  obj: 'quu',
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";

_gbd(
  {
    multi: "foo",
    line: "bar",
    obj: "quu",
  },
  "{\\n  multi: 'foo',\\n  line: 'bar',\\n  obj: 'quu'\\n}"
);


`;

exports[`macros 3. macros: 3. macros 2`] = `

import { gbd } from 'gbd.macro'
gbd({
  multi: 'foo',
  line: 'bar',
  obj: 'quu',
});

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

(0, _gbd2.gbd)(
  {
    multi: "foo",
    line: "bar",
    obj: "quu",
  },
  "{\\n  multi: 'foo',\\n  line: 'bar',\\n  obj: 'quu'\\n}"
);


`;

exports[`macros 4. macros: 4. macros 1`] = `

import { gbd } from 'gbd.macro'
function test() {
  gbd({
    multi: 'foo',
    line: 'bar',
    obj: 'quu',
  });
}

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";

function test() {
  _gbd(
    {
      multi: "foo",
      line: "bar",
      obj: "quu",
    },
    "{\\n  multi: 'foo',\\n  line: 'bar',\\n  obj: 'quu'\\n}"
  );
}


`;

exports[`macros 4. macros: 4. macros 2`] = `

import { gbd } from 'gbd.macro'
function test() {
  gbd({
    multi: 'foo',
    line: 'bar',
    obj: 'quu',
  });
}

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

function test() {
  (0, _gbd2.gbd)(
    {
      multi: "foo",
      line: "bar",
      obj: "quu",
    },
    "{\\n  multi: 'foo',\\n  line: 'bar',\\n  obj: 'quu'\\n}"
  );
}


`;

exports[`macros 5. macros: 5. macros 1`] = `

import { gbd } from 'gbd.macro'
function test() {
  gbd(1 + 2);
}
gbd(1 + 2);

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";

function test() {
  _gbd(1 + 2, "1 + 2");
}

_gbd(1 + 2, "1 + 2");


`;

exports[`macros 5. macros: 5. macros 2`] = `

import { gbd } from 'gbd.macro'
function test() {
  gbd(1 + 2);
}
gbd(1 + 2);

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

function test() {
  (0, _gbd2.gbd)(1 + 2, "1 + 2");
}

(0, _gbd2.gbd)(1 + 2, "1 + 2");


`;

exports[`macros 6. macros: 6. macros 1`] = `

import "./styles.css";
import { gbd } from "gbd.macro";

gbd(1 + 2);
gbd(99 / 3);

      ↓ ↓ ↓ ↓ ↓ ↓

import { gbd as _gbd } from "gbd";
import "./styles.css";

_gbd(1 + 2, "1 + 2");

_gbd(99 / 3, "99 / 3");


`;

exports[`macros 6. macros: 6. macros 2`] = `

import { gbd } from "gbd.macro";

gbd(1 + 2);
gbd(99 / 3);

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var _gbd2 = require("gbd");

(0, _gbd2.gbd)(1 + 2, "1 + 2");
(0, _gbd2.gbd)(99 / 3, "99 / 3");


`;
